pkg/processor/trigger/http/cors/cors.go

```diff
@@ -0,0 +1,106 @@
+package cors
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/nuclio/nuclio/pkg/common"
+
+	"github.com/valyala/fasthttp"
+)
+
+type CORS struct {
+	Enabled bool
+
+	// allow configuration
+	AllowOrigin      string
+	AllowMethods     []string
+	AllowHeaders     []string
+	AllowCredentials bool
+
+	// preflight
+	PreflightRequestMethod string
+	PreflightMaxAgeSeconds int64
+
+	// computed
+	allowMethodsStr           string
+	allowHeadersStr           string
+	preflightMaxAgeSecondsStr string
+	allowCredentialsStr       string
+	simpleMethods             []string
+}
+
+func NewCORS() *CORS {
+	return &CORS{
+		Enabled:     true,
+		AllowOrigin: "*",
+		AllowMethods: []string{
+			fasthttp.MethodHead,
+			fasthttp.MethodGet,
+			fasthttp.MethodPost,
+			fasthttp.MethodPut,
+			fasthttp.MethodDelete,
+			fasthttp.MethodOptions,
+		},
+		AllowHeaders: []string{
+			fasthttp.HeaderAccept,
+			fasthttp.HeaderContentLength,
+			fasthttp.HeaderContentType,
+
+			// nuclio custom
+			"X-nuclio-log-level",
+		},
+		AllowCredentials:       false,
+		PreflightRequestMethod: fasthttp.MethodOptions,
+		PreflightMaxAgeSeconds: -1, // disable cache by default
+	}
+}
+
+func (c *CORS) IsSetAndMatchOrigin(origin string) bool {
```

Comment:
Why not just call all of these: `OriginAllowed`, `MethodAllowed`, `HeadersAllowed`?